### PR TITLE
Feature/45 creates score card

### LINKED ISSUE #45: Score card && UI between rounds, handled by RoundController
**Description**
When the RoundTimer is up the RoundController will enable a UI, pause game, and wait for input from all players before letting next round start.

**Subtasks**
- [ ] Creates UI elements and function for enabling/disabling it, handled by RoundController
- [ ] Pauses game while in menu.
- [ ] Starts new round after getting input.

### PR DESCRIPTION
**Pull request by DreamingJohnny**
Closes #45 

**Description**  
Set up a basic between rounds UI that kicks of when the player(s) win a round

### CODE REVIEW

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 1]
```diff
@@ -25,11 +31,13 @@ public class RoundController : MonoBehaviour {
 		private void Update() {
 			roundTimer = Mathf.Max(0, roundTimer - Time.deltaTime);
 
-			if (roundTimer <= 0) {
-				currentRound++;
-				SendNextDifficulty();
-				roundTimer = roundDuration;
-			}
+			if (roundTimer <= 0) RoundWon();
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/RoundController.cs:34]: Would this be a fitting place to stop the viking controller spawning in more vikings?

File: Assets/00_Content/Scripts/Rounds/ScoreCard.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		//TODO: Get number of active players,
+		//TODO: Enable that many icons on UI
+		//TODO: Wait for input from each player
+		//TODO: Have their icon switch over when they give correct input
+		//TODO: Have the icon switch back if player gives same input again.
+		//TODO: When all have given input start next round.
+	}
+
+	public void OnEnable() {
+		Time.timeScale = 0f;
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/ScoreCard.cs:24]: Is it really optimal to stop the time between rounds?

File: Assets/00_Content/Scripts/Rounds/ScoreCard.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+public class ScoreCard : MonoBehaviour {
+	[SerializeField] private GameObject scoreCardUI;
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/ScoreCard.cs:8]: This seems to be referring to itself, can't you use `gameobject` instead?

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 4]
```diff
@@ -25,11 +31,13 @@ public class RoundController : MonoBehaviour {
...
+		}
+
+		private void RoundWon() {
+			SetUpNextRound();
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/RoundController.cs:38]: Would it be better to just deactivate the round timer here and setup the next round when returning from the scorecard?

**Review by MrJackey — CHANGES_REQUESTED** [review]
I have some thoughts, some may be due to me having a different picture in my head. Also rescaling the canvas might be a good idea, I was unable to see everything when I tested it.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 5]
```diff
@@ -8,15 +10,19 @@ public class RoundController : MonoBehaviour {
 		[SerializeField] private ScalingData[] playerDifficulties;
 		[SerializeField] private Tavern tavern;
 		[SerializeField] private VikingController vikingController;
+		[SerializeField] private GameObject scoreCardUI;
```
[the remark this answers is not in the document]
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs]: Fixed.
RoundController now spawns ScoreCanvas and references the script directly.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs (hunk 1 above)
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs:36]: Fixed.

File: Assets/00_Content/Scripts/Rounds/ScoreCard.cs (hunk 2 above)
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/ScoreCard.cs]: I've removed it, still needs to work on the problems with the input systems though.

File: Assets/00_Content/Scripts/Rounds/ScoreCard.cs (hunk 3 above)
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/ScoreCard.cs]: Fixed

**Comment by DreamingJohnny** [comment]
Jonatan is going to look @ the bugs that has to do with the new input system later so I'm handing this over to him.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 6]
```diff
@@ -10,33 +11,44 @@ public class RoundController : SingletonBehaviour<RoundController> {
...
 		[SerializeField] private Tavern tavern;
 		[SerializeField] private VikingController vikingController;
+		[SerializeField] private ScoreCard scoreCardPrefab;
 		[SerializeField, Tooltip("seconds/round")]
 		private int roundDuration;
 
+		private ScoreCard scoreCard;
 		private float roundTimer;
 		private int currentRound = 1;
+		private bool hasWon;
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/RoundController.cs:21]: Would roundIsActive or something similar be a more descriptive name? It feels a bit weird toggling if you have won during the same session.

File: Assets/00_Content/Prefabs/Vikings/VikingController.prefab [hunk 7]
```diff
@@ -16,7 +16,7 @@ GameObject:
   m_Icon: {fileID: 0}
   m_NavMeshLayer: 0
   m_StaticEditorFlags: 0
-  m_IsActive: 1
+  m_IsActive: 0
```
Comment (MrJackey) [inline Assets/00_Content/Prefabs/Vikings/VikingController.prefab:19]: The vikingcontroller prefab probably shouldn't be deactivated right?

**Review by MrJackey — COMMENTED** [review]
Teeny tiny thoughts

File: Assets/00_Content/Scripts/Rounds/RoundController.cs (hunk 6 above)
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs]: Mhm yeah, much better, I've renamed it to "isRoundActive"

**Review by MrJackey — APPROVED** [review]
:shipit:

**Pull request merged**